docker-app/qfieldcloud/core/models.py

```diff
@@ -144,6 +158,13 @@ class UserAccount(models.Model):
     is_email_public = models.BooleanField(default=False)
     avatar_uri = models.CharField(_("Profile Picture URI"), max_length=255, blank=True)
 
+    notifs_frequency = models.CharField(
+        max_length=25,
+        verbose_name=_("Email frequency for notifications"),
+        choices=NOTIF_CHOICES,
+        default=NOTIF_DAILY,
+    )
```

Comment:
Is it possible to have this a configuration like `notifs_frequency_m = models.UIntField(min=0, choices=IntChoices)`? This way it would be easier to change the configuration for the users.

Reply:
Currently there's just four different django_crons set up (SendNotificationsJobImmediately, SendNotificationsJobHourly, SendNotificationsJobDaily, SendNotificationsJobWeekly).
If we wanted this to be customizable, we'd run every minute for everybody, but would then need to store last execution time for each user.

Not sure it's worth going down that rabbit hole ?

Reply:
hm, isn't it enough to have a `sent_at` field in the `Notification` model? Whenever the cron runs, you call:
```
notifs = Notification.objects.filter(user=user, sent_at__isnull=True)
send_notif_emails(notifs)
```
Or you mean that we don't have `sent_at` field at all?

Reply:
The issue is that if we wanted to make it easy to customize the frequency using an integer (minutes) rather than predefined choices, we'd also need some kind of scheduling strategy.

Currently it uses 4 different crons, on for each frequency, so we have predefined/hard-coded available schedules only.

It's not impossible to make it customizable, but then we'd need to store when the cron was last sending emails run for every user as they'd all run on the same minutely cron. IMO not really worth the complexity.

We could add more predefine frequencies if you think it's needed, but I think with immediate (minute), hourly, daily and weekly we've got most need covered ?

Reply:
Ok as per discussion implemented this (hehe not sure why this was overcomplex in my head ?). Used a DurationField instead of int field so it simplifies date arithmetic functions